skbio/tree/_bme.py

```diff
@@ -0,0 +1,352 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from operator import itemgetter
+from skbio.tree import TreeNode
+from skbio.tree._gme import _average_distance_k, _average_distance_k_upper
+
+
+def bme(dm, allow_edge_estimation=True):
+    r"""Perform balanced minimum evolution (BME) for phylogenetic reconstruction.
+
+    Parameters
+    ----------
+    dm : skbio.DistanceMatrix
+        Input distance matrix containing distances between taxa.
+    allow_edge_estimation : bool, optional
+        Whether to perform an OLS-based estimation of edge values (``True``, default)
+        or return a tree without edge values assigned (``False``).
+
+    Returns
+    -------
+    TreeNode
+        Reconstructed phylogenetic Tree with estimated edge
+        values (if ``allow_edge_estimation`` is ``True``).
+
```

Comment:
Maybe add a "See Also" section with `nj` and `gme`. Also add `bme` to the "See Also" section of `nj`.